compiler/src/dotty/tools/dotc/printing/Formatting.scala

```diff
@@ -86,7 +86,8 @@ object Formatting {
         case hb: HighlightBuffer =>
           hb.toString
         case str: String if ctx.settings.color.value != "never" =>
-          new String(SyntaxHighlighting(str).toArray)
+          if (str.matches(".*\u001b\\[.*?m.*")) str // was already colored by some other printer
```

Comment:
This seems very expensive. Error messages are colored because of the `hl` string interpolator. The interpolator call show on its arguments. Can we instead disable syntax highlighting when calling show for the argument of the interpolator?

Reply:
Redefining `hl` as:
```scala
def hl(args: Any*)(implicit ctx: Context): String = {
  val nestedCtx = ctx.fresh.setSetting(ctx.settings. color, true)
  new SyntaxFormatter(sc).assemble(args)(nestedCtx).stripMargin
)
```
should work. I am wondering how expensive is `ctx.fresh` though.

Reply:
This might work if the argument in `args` is still of type `Text`.

Reply:
At this point it is too late. I created instead an `AlreadyColored` that fixes the issue for each specific case. It should be added to any printed tree that is already colored.